app/src/main/java/org/eyeseetea/malariacare/BaseActivity.java

```diff
@@ -114,13 +114,16 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onRequestPermissionsResult(int requestCode,
             String permissions[], int[] grantResults) {
         if (Permissions.processAnswer(requestCode, permissions, grantResults)) {
-            EyeSeeTeaApplication.permissions.requestNextPermission();
+            EyeSeeTeaApplication.permissions.requestNextPermission(this);
             if (EyeSeeTeaApplication.permissions.areAllPermissionsGranted()) {
                 PhoneMetaData phoneMetaData = getPhoneMetadata();
                 Session.setPhoneMetaData(phoneMetaData);
             }
         } else {
-            onDestroy();
+            Intent homeIntent = new Intent(Intent.ACTION_MAIN);
```

Comment:
I think that this code provoke a activity refresh first time when you navigate to dashboard from login

Reply:
i tested it and this code is only executed when the user don't accept the permissions